etc/env/example.env

```diff
@@ -197,3 +201,81 @@ DATASETS_SERVICE_PORT=8880
 AUTHORIZATION_SERVICE_VERSION=v0.0.1-alpha
 AUTHORIZATION_SERVICE_HOST=0.0.0.0
 AUTHORIZATION_SERVICE_PORT=7000
+
+
+# keycloak service
+CANDIG_AUTH_CONTAINER_NAME=candigauth.local
+KEYCLOAK_VERSION=9.0.2
+KC_ADMIN_USER=
+KC_ADMIN_PW=
+KC_TEST_USER=
+KC_TEST_PW=
+KC_TEST_USER_TWO=
+KC_TEST_PW_TWO=
+KC_REALM=candig
+KC_CLIENT_ID=cq_candig
+KC_LOGIN_REDIRECT_PATH=/auth/login
+## KC_SECRET= ###(generated in setup.sh)
+## KC_CLIENT_ID_64= ###(generated in setup.sh)
+## KC_PUBLIC_KEY= ###(generated in setup.sh)
+KEYCLOAK_SERVICE_PUBLIC_PORT=8081
+KEYCLOAK_SERVICE_CONTAINER_PORT=8081
+KEYCLOAK_SERVICE_HOST=0.0.0.0
+
+KEYCLAOK_SERVICE_PUBLIC_PROTO=http
+KEYCLAOK_SERVICE_PRIVATE_PROTO=http
+KEYCLOAK_SERVICE_PUBLIC_URL=${KEYCLAOK_SERVICE_PUBLIC_PROTO}://${CANDIG_AUTH_DOMAIN}:${KEYCLOAK_SERVICE_PUBLIC_PORT}
+KEYCLOAK_SERVICE_PRIVATE_URL=${KEYCLAOK_SERVICE_PRIVATE_PROTO}://${CANDIG_AUTH_CONTAINER_NAME}:${KEYCLOAK_SERVICE_CONTAINER_PORT}
+
+
+# tyk service
+CANDIG_SERVER_CONTAINER_NAME=candig_server_candig-server_1
+CANDIG_SERVER_PUBLIC_PORT=8080
+TYK_VERSION=v2.9.3.1
+TYK_REDIS_VERSION=4.0.14-alpine
+TYK_SERVICE_CONTAINER_NAME=tyk
+TYK_SERVICE_PUBLIC_PORT=8080
+TYK_SERVICE_HOST=0.0.0.0
+TYK_SERVICE_PRIVATE_URL=http://${TYK_SERVICE_CONTAINER_NAME}:${TYK_SERVICE_PUBLIC_PORT}
+TYK_TARGET_URL=http://${CANDIG_ARBITER_HOST}:${CANDIG_ARBITER_SERVICE_PORT}
+TYK_LOGIN_TARGET_URL=http://${TYK_SERVICE_HOST}:${TYK_SERVICE_PUBLIC_PORT}
+TYK_LISTEN_PATH=
+TYK_POLICY_ID=candig_policy
+
+## -- generic template stuff
+AUTH_API_ID=11
+AUTH_API_NAME=Authentication
+AUTH_API_SLUG=authentication
+
+CANDIG_API_ID=21
+CANDIG_API_NAME=CanDIG
+CANDIG_API_SLUG=candig
+CANDIG_PUBLIC_URL=http://${CANDIG_DOMAIN}:${CANDIG_SERVER_PUBLIC_PORT}
+## --
+
+
+# vault service
+VAULT_CONTAINER_NAME=vault
+VAULT_FILE_PATH="/vault/data"
+VAULT_TLS_DISABLE=1
+VAULT_UI=true
+VAULT_SERVICE_PORT=8200
+VAULT_SERVICE_HOST=0.0.0.0
+VAULT_SERVICE_PUBLIC_URL=http://${VAULT_SERVICE_HOST}:${VAULT_SERVICE_PORT}
+VAULT_SERVICE_URL=http://${VAULT_CONTAINER_NAME}:${VAULT_SERVICE_PORT}
+## VAULT_JWKS= ###(generated in setup.sh)
+
+
+# OPA
+OPA_VERSION=latest
+OPA_CONTAINER_NAME=opa
+OPA_PORT=8181
+OPA_LOG_LEVEL=debug
+OPA_URL=http://${OPA_CONTAINER_NAME}:${OPA_PORT}
+
+CANDIG_AUTHZ_SERVICE_PORT=8182
+
+# same as former candig-server port - used to relay to new c-s port
+CANDIG_ARBITER_SERVICE_PORT=3002
```

Comment:
The arbiter service is ready, and I think it should be shipped.

If we remodel this PR, it should be renamed to AuthN instead of AuthZ, as only components like Tyk, Keycloak and Vault would be involved.

On behalf of this comment, though, I will update the main OPA rego file to not apply the rules used in the demo (allowing entry for one user, but deny access for another)

Reply:
Oh no worries! I was not sure arbiter was working. This is great!  I would like to see the rules that are applied and how they are applied. This is pretty interesting.